icu4j/main/tests/core/src/com/ibm/icu/dev/test/util/ULocaleTest.java

```diff
@@ -5175,21 +5177,20 @@ public void TestCanonical() {
         // also test with script, variants and extensions
         Assert.assertEquals("fa-Cyrl-AF-1009-u-ca-roc", canonicalTag("prs-Cyrl-1009-u-ca-roc"));
 
-        if (!logKnownIssue("21236", "skip some canonicalization tests until code fixed")) {
-            //  language _ country -> language _ script _ country
-            Assert.assertEquals("pa-Guru-IN", canonicalTag("pa-IN"));
-        }
+        //  language _ country -> language _ script _ country
```

Comment:
This comment does not match the tested behavior. Please update or remove the comment.

Do you know/remember why we expected canonicalization to add script subtags?

Reply:
I start to write the code and test tests when Mark and I reach the understanding of how the algorithm should work but before he change the data in CLDR. After I wrote these test based on the CLDR shipped with ICU 67, he figured some of the CLDR data were there fore addLikelySubtag and do not need to be there so he removed them. So now this part won't change since those entries got removed but I forgot to change the comments.